dash/testing/plugin.py

```diff
@@ -0,0 +1,50 @@
+# pylint: disable=missing-docstring
+import pytest
+
+from selenium import webdriver
+
+from dash.testing.application_runners import ThreadedRunner, ProcessRunner
+from dash.testing.browser import Browser
+
+WEBDRIVERS = {
+    "Chrome": webdriver.Chrome,
+    "Firefox": webdriver.Firefox,
+    "Remote": webdriver.Remote,
+}
+
+
+def pytest_addoption(parser):
+    # Add options to the pytest parser, either on the commandline or ini
+    # TODO add more options for the selenium driver.
+    dash = parser.getgroup("Dash", "Dash Integration Tests")
+
+    dash.addoption(
+        "--webdriver",
+        choices=tuple(WEBDRIVERS.keys()),
+        default="Chrome",
+        help="Name of the selenium driver to use",
+    )
+
+###############################################################################
+# Fixtures
+###############################################################################
+
+
+@pytest.fixture
+def thread_server():
+    """Start a local dash server in a new thread"""
+    with ThreadedRunner() as starter:
+        yield starter
+
+
+@pytest.fixture
+def process_server():
+    """Start a Dash server with subprocess.Popen and waitress-serve"""
+    with ProcessRunner() as starter:
+        yield starter
+
+
+@pytest.fixture
+def br(request):
```

Comment:
that might be a little too terse, and could be confused with `<br>` - perhaps `browser`?

Reply:
I used to call that `wd` for webdriver, but found now `br` is nicer. I get your concern about confusion. 

in our test case context, it's less serious as the html component will be `html.Br`, and the main reason I would fight for being terse is we will use fixtures everywhere, and I'm kinda finger saver

Reply:
right, due to pytest magic this becomes the var name people will need to use throughout their tests - eg `br.wait_for_element_by_css_selector(...)` - though TBH it seems a bit funny to worry about the length of the word `browser` when we have methods like `wait_for_element_by_css_selector` 😏

I could be overruled if other folks are on board with `br`, but personally I'd still prefer `browser`.

Reply:
I kept the `wait_for_element_by_css_selector` for back compatibilities, we should use css selector by default, so that can actually be just `wait_for_element` and you remind me to add the default one.